apt/private/version_constraint.bzl

```diff
@@ -57,29 +68,26 @@ def _parse_depends(depends_raw):
 
     return depends
 
-def _version_relop(va, vb, op):
-    if op == "<<":
-        return version_lib.lt(va, vb)
-    elif op == ">>":
-        return version_lib.gt(va, vb)
-    elif op == "<=":
-        return version_lib.lte(va, vb)
-    elif op == ">=":
-        return version_lib.gte(va, vb)
-    elif op == "=":
-        return version_lib.eq(va, vb)
-    fail("unknown op %s" % op)
-
-def _is_satisfied_by(va, vb):
-    if vb[0] != "=":
-        fail("Per https://www.debian.org/doc/debian-policy/ch-relationships.html only = is allowed for Provides field.")
-
-    return _version_relop(va[1], vb[1], va[0])
+def _parse_provides(provides_raw):
+    provides = _parse_dep(provides_raw)
+
+    if not provides["version"]:
+        return provides
+
+    op, version = provides["version"]
+
+    if op != "=":
```

Comment:
we can't do this, is_satisfied_by checks to see if two different version constraint satisfied where as relop just checks two different versions by an operator.

Reply:
I just moved the validation to the parsing so that then, we know that all versions in Provides have `=` and thus, we can use the same logic as in relop. Also, doing this makes the code for virtual packages in resolve to be pretty much the same as the real packages.

Reply:
I'm going to bring back `_is_satisfied_by`, leaving `_parse_provides` so that we get both improvements (the validation at parsing and the semantics and validation of `_is_satisfied_by` at the time of comparison.